src/main/scala/linting/rule/LintConflictingModuleNames.scala

```diff
@@ -0,0 +1,101 @@
+// See LICENSE.SiFive for license details.
+
+package freechips.rocketchip.linting
+package rule
+
+import firrtl._
+import firrtl.annotations.{IsModule, SingleTargetAnnotation, Target}
+import firrtl.ir._
+import firrtl.options.Dependency
+
+import chisel3.aop.{Aspect, Select}
+import chisel3.RawModule
+
+import scala.collection.mutable
+
+/** Captures the original desired name for a module
+  */
+case class DesiredNameAnnotation(
+  desiredName: String,
+  target: IsModule
+) extends SingleTargetAnnotation[IsModule] {
+  def duplicate(newTarget: IsModule): DesiredNameAnnotation = {
+    this.copy(target = newTarget)
+  }
+}
+
+/** Collects [[DesiredNameAnnotation]]s for [[LintConflictingModuleNames]] to lint
+  */
+case object LintConflictingModuleNamesAspect extends Aspect[RawModule] {
+  def toAnnotation(top: RawModule): AnnotationSeq = {
+    Select.collectDeep(top) {
+      case m: RawModule => DesiredNameAnnotation(m.desiredName, m.toTarget)
+    }.toSeq
+  }
+}
+
+/** This LintRule checks for module name collisions
+  *
+  * Module name collisions occur when different [[Module]]s are annotated with
+  * [[DesiredNameAnnotation]]s that have the same desiredName. Module name
+  * conflicts will cause a [[LintViolation]].
+  */
+final class LintConflictingModuleNames extends LintRule {
+  val recommendedFix: String = "override desiredName based on module parameters ('override def desiredName = \"...\"')"
```

Comment:
Or use RenameModuleAspect, correct?

Reply:
Can you. update the recommended fix to include the aspect option?

Reply:
@albertchen-sifive this is the only thing it looks like you did not do yet. @azidar are you OK merging without this?

Reply:
this was updated

Reply:
really? I still see only the def desiredName part, I think @azidar was hoping for something like "or use an aspect to apply OverrideDesiredName annotation"

Reply:
https://github.com/chipsalliance/rocket-chip/pull/2452/files#diff-0b36929ef6150dd05202cc16819b9bb2R46

Reply:
yep, now I see it!